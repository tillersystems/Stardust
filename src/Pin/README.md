# Pin

### Usage

```jsx
import { Pin } from '@tillersystems/stardust';
```

<!-- STORY -->

### Properties

- `color` - Pin color.
- `height` - Pin height.
- `width` - Pin width.

| propName | propType | defaultValue | isRequired |
| -------- | -------- | ------------ | ---------- |
| `color`  | `string` |              | \*         |
| `height` | `string` | `1rem`       | -          |
| `width`  | `string` | `1rem`       | -          |

### Example

```jsx
import { Pin } from '@tillersystems/stardust';

render() {
  return (
    <Pin color="red" width='3rem' height='3rem' />
  )
}
```
