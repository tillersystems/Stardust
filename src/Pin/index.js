import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Defines a pin component (a little round with color).
 *
 * @param {string} width - The width of the pin.
 * @param {string} height - The height of the pin.
 * @param {string} color - The color of the pin.
 *
 * @return {jsx}
 */
const Pin = ({ className }) => <div className={className} />;

/** Prop types. */
const { string } = PropTypes;
/* eslint-disable react/no-unused-prop-types */
Pin.propTypes = {
  className: string,
  width: string,
  height: string,
  color: string.isRequired,
};

/** Default props. */
Pin.defaultProps = {
  className: '',
  width: '1rem',
  height: '1rem',
};

const styledPin = styled(Pin)`
  width: ${({ width }) => width || '1rem'};
  height: ${({ height }) => height || '1rem'};

  border-radius: 50%;

  background: ${({ color }) => color};
`;
styledPin.displayName = 'Pin';

export default styledPin;
