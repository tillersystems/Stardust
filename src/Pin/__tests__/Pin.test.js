import React from 'react';

import Pin from '..';

describe('<Pin />', () => {
  test('should render pin without a problem', () => {
    const { container } = render(<Pin color="blue" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render with custom size', () => {
    const size = '3rem';
    const { container } = render(<Pin color="blue" height={size} width={size} />);

    expect(container.firstChild).toHaveStyleRule('height', size);
    expect(container.firstChild).toHaveStyleRule('width', size);
  });
});
