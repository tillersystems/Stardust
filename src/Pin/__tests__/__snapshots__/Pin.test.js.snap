// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pin /> should render pin without a problem 1`] = `
.c0 {
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  background: blue;
}

<div
  class="c0"
/>
`;
